src/lib/src/component/navigation/navigation.component.spec.ts

```diff
@@ -364,18 +371,61 @@ describe('NavigationComponent', () => {
   it('invokes test execution for currently selected test file when "run" button is clicked', async(() => {
     // given
     createRootWithSubfolder();
+    component.uiState.selectedElement = tclFile;
     fixture.detectChanges();
     let runIcon = sidenav.query(By.css('#run'));
-    component.uiState.selectedElement = listedFile;
     resetCalls(executionService);
 
     // when
     runIcon.nativeElement.click();
 
     // then
     fixture.whenStable().then(() => {
-      verify(executionService.execute(listedFile.path)).once();
-      expect(listedFile.state).toEqual(ElementState.Running);
+      verify(executionService.execute(tclFile.path)).once();
+      expect(tclFile.state).toEqual(ElementState.Running);
     });
   }));
+
+  it('disables the run button when selecting a non-executable file', async(() => {
+    // given
+    component.workspace = new Workspace({
+      name: 'root',
+      path: '',
+      type: ElementType.Folder,
+      children: [ nonExecutableFile, tclFile ]
+    });
+    fixture.detectChanges();
+    let runIcon = sidenav.query(By.css('#run'));
+    component.uiState.selectedElement = tclFile;
+
+    // when
+    component.selectElement(nonExecutableFile.path)
+
+    // then
+    fixture.whenStable().then(() => {
+      expect(runIcon.properties['disabled']).toBeTruthy;
+    });
+  }));
+
+  it('enables the run button when selecting an executable file', async(() => {
+    // given
+    component.workspace = new Workspace({
+      name: 'root',
+      path: '',
+      type: ElementType.Folder,
+      children: [ nonExecutableFile, tclFile ]
+    });
+    fixture.detectChanges();
```

Comment:
This code and the one above is redundant. We should probably extract this into a separate method.